src/alchemlyb/estimators/ti_gaussian_quadrature_.py

```diff
@@ -0,0 +1,228 @@
+            
+            adelta += np.diagflat(np.array(out), k=j)
+            ad_delta += np.diagflat(np.array(dout), k=j)
+
+        # yield standard delta_f_ cumulative free energies from one state to another
+        self._delta_f_ = pd.DataFrame(
+            adelta, columns=means_new.index.values, index=means_new.index.values
+        )
+        
+        # yield standard deviation d_delta_f_ between each state
+        self._d_delta_f_ = pd.DataFrame(
+            np.sqrt(ad_delta),
+            columns=variances_new.index.values,
+            index=variances_new.index.values,
+        )
+        
+        self.dhdl = self.means
+        self.dhdl.attrs = dHdl.attrs
+        self._states_ = means_new.index.values.tolist()
+        self._delta_f_.attrs = dHdl.attrs
+        self._d_delta_f_.attrs = dHdl.attrs
+
+        return self
+
+    def separate_dhdl(self):
+        """
+        For transitions with multiple lambda, the attr:`dhdl` would return
+        a :class:`~pandas.DataFrame` which gives the dHdl for all the lambda
+        states, regardless of whether it is perturbed or not. This function
+        creates 3 lists of :class:`numpy.array`, :class:`pandas.Series` and 
+        :class:`pandas.Series` for each lambda, where the lists describe 
+        the lambda values, potential energy gradient and variance values for
+        the lambdas state that is perturbed.
+
+        Returns
+        ----------
+        lambda_list : list
+            A list of :class:`numpy.array` such that ``lambda_list[k]`` is the
+            lambda values with respect to each type of lambda.
+        dhdl_list : list
+            A list of :class:`pandas.Series` such that ``dHdl_list[k]`` is the
+            potential energy gradient with respect to lambda for each
+            configuration that lambda k is perturbed.
+        variance_list : list
+            A list of :class:`pandas.Series` such that ``variance_list[k]`` is the
+            variance of the potential energy gradient with respect to lambda for each
+            configuration that lambda k is perturbed.
+        """
+        lambda_list = []
+        dhdl_list = []
+        variance_list = []
+        # get the lambda names
+        l_types = self.means.index.names
+        # get the lambda vaules
+        lambdas = self.means.reset_index()[self.means.index.names].values
+
+        if len(self.means.index.names) == 1:
+            name = self.means.columns[0]
+            lambda_list.append(self.means.index)
+            dhdl_list.append(self.means[name])
+            variance_list.append(self.variances[name])
+
+        else:
+            # simultanouesly scaling of multiple lambda types are not supported
+            if (((0.0 < lambdas) & (lambdas < 1.0)).sum(axis=1) > 1.0).any():
+                raise ValueError('The lambda values for gaussian quadrature are not supported, please use trapezoidal rule instead.')
```

Comment:
Please make this error message more clear. 
Also `(((0.0 < lambdas) & (lambdas < 1.0)).sum(axis=1) > 1.0).any()` is a bit confusing to read. Could you give some failed test examples in the test so I could see when this will trigger?

Reply:
I have changed the error message.
`(((0.0 < lambdas) & (lambdas < 1.0)).sum(axis=1) > 1.0).any()` is to make sure that in the case of multiple lambda, only one lambda could be scaled at a time, for example (0.0469, 0.0), (0.2308, 0.0), ..., (0.9531, 0. 0). Otherwise, the sum would be greater than 1 and trigger that error.
There is a test example (`test_TI_GQ_multi_lambda_scaling(ethanol_Coulomb)`) where a case was created artificially by scaling both lambdas at the same time, for example (0.0469, 0.0469), (0.2308, 0.2308), ..., (0.9531, 0.9531). This lambda schedule might be problematic when applied in real simulations. The error would be caught by `pytest.raise`.